### PR TITLE
remove alias of neutral fields to generic names

### PR DESCRIPTION
**Pull request by brittonsmith**
In yt-4.0, the `X_number_density` and related fields have been re-defined to be aliases to `X_nuclei_density` instead of `X_p0_number_density`. Hence, we need to drop that old aliasing here, too. This PR removes the generic alias names and requires using `p0` fields for neutral species. I think I've gotten all the right places, but am not sure, so I'll wait for some feedback on this one.

### CODE REVIEW

**Comment by brittonsmith** [comment]
Most of the failing tests were looking for the field names that have now been removed. Two were enzo answer tests that were using the old field names incorrectly. I'm updating the gold standard for them now.

File: .circleci/config.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ commands:
             echo 'TRIDENT_CONFIG=$HOME/.trident/config.tri' >> $BASH_ENV
             echo 'YT_GOLD=8dcc2fa135dfe7a62438507066527caf5cb379c1' >> $BASH_ENV
             echo 'YT_HEAD=yt-4.0' >> $BASH_ENV
-            echo 'TRIDENT_GOLD=test-standard-sph-viz-v6' >> $BASH_ENV
+            echo 'TRIDENT_GOLD=HEAD' >> $BASH_ENV
```
Comment (brittonsmith) [inline .circleci/config.yml:16]: After merging, I will push another commit directly to master changing this to `test-standard-sph-viz-v7` and push a tag with that name.

**Comment by brittonsmith** [comment]
@chummels, I think it would be a good idea for you to look this over before it's merged.

**Comment by chummels** [comment]
I've found a number of places that you missed.  I've been committing local changes to address these, but I ran into some problems with running the tests on the current gold standard, which I addressed in Issue #133 , because I've been getting a large number of failures locally and don't want to commit garbage to that messes up this PR.

File: .circleci/config.yml [hunk 2]
```diff
@@ -70,7 +70,7 @@ commands:
           command: |
             source $BASH_ENV
             source $HOME/venv/bin/activate
-            git checkout ${<< parameters.tridenttag >>}
+            git checkout ${<< parameters.tridenttag >>} || git checkout $TRIDENT_HEAD
```
Comment (brittonsmith) [inline .circleci/config.yml:73]: With this change, we won't have to first set `TRIDENT_GOLD` to be `HEAD` and then change it with a direct commit after the pull request is merged. We'll still have to remember to add the tag and push it.

**Comment by brittonsmith** [comment]
@chummels, does anything else need to be done for this?

**Comment by chummels** [comment]
One thing I noticed is that yt-4.0 doesn't seem to have made `H_number_density` alias to `H_nuclei_density`, at least not for the Enzo datasets.  It just seems like one gets different behavior with different frontends.  Do you think it's worth explicitly mapping `H_number_density` to `H_nuclei_density` for all frontends in yt4 so there isn't any random behavior?

**Comment by brittonsmith** [comment]
Looks like tests are passing. I have to go now, but feel free to merge when ready.

**Pull request merged**